homeassistant/core.py

```diff
@@ -1214,29 +1215,57 @@ def call(
             context=context,
         )
 
+        coro = self._execute_service(handler, service_call)
         if not blocking:
-            self._hass.async_create_task(self._safe_execute(handler, service_call))
+            self._run_service_in_background(coro, service_call)
             return None
 
+        task = self._hass.async_create_task(coro)
         try:
-            async with timeout(limit):
-                await asyncio.shield(self._execute_service(handler, service_call))
```

Comment:
So the biggest change here is that we're no longer shielding things. I suggest we merge this as-is and see how it highlights any issues before 109 is released. If we find some issues, we can decide to apply shields to the scenarios that are failing (thinking Rest/Websocket call service APIs)